asm-lsp/lsp.rs

```diff
@@ -1310,12 +1356,17 @@ pub fn get_sig_help_resp(
             if caps.len() == 1 && caps[0].node.end_byte() < curr_doc.len() {
                 if let Ok(instr_name) = caps[0].node.utf8_text(curr_doc) {
                     let mut value = String::new();
+                    // Switch to a better structure
                     let mut has_x86 = false;
                     let mut has_x86_64 = false;
                     let mut has_z80 = false;
                     let mut has_arm = false;
-                    let (x86_info, x86_64_info, z80_info, arm_info, riscv_info) =
-                        search_for_hoverable_by_arch(instr_name, instr_info);
+                    let mut has_arm64 = false;
+                    // ensure hovered instruction is always lowercase
+                    let hovered_instr_name = instr_name.to_ascii_lowercase();
+                    let (x86_info, x86_64_info, z80_info, arm_info, arm64_info, riscv_info) =
```

Comment:
yeah, this kinda code has really grown out of hand. I'm looking forward to cleaning it up in #158. Would really appreciate your input there once it's been cleaned up a little :)

Reply:
Definitely looking forward to overhauling some inappropriate data structures used all around the codebase.